nipype/pipeline/engine/nodes.py

```diff
@@ -510,49 +510,55 @@ def _get_hashval(self):
         return self._hashed_inputs, self._hashvalue
 
     def _get_inputs(self):
-        """Retrieve inputs from pointers to results file
+        """
+        Retrieve inputs from pointers to results files.
 
         This mechanism can be easily extended/replaced to retrieve data from
         other data sources (e.g., XNAT, HTTP, etc.,.)
         """
         if self._got_inputs:
             return
 
-        logger.debug('Setting node inputs')
+        prev_results = defaultdict(list)
         for key, info in list(self.input_source.items()):
-            logger.debug('input: %s', key)
-            results_file = info[0]
-            logger.debug('results file: %s', results_file)
-            outputs = _load_resultfile(results_file).outputs
+            prev_results[info[0]].append((key, info[1]))
+
+        logger.debug('[Node] Setting %d connected inputs from %d previous nodes.',
+                     len(self.input_source), len(prev_results))
+
+        for results_fname, connections in list(prev_results.items()):
+            outputs = None
+            try:
+                outputs = _load_resultfile(results_fname).outputs
+            except AttributeError as e:
+                logger.critical('%s', e)
+
             if outputs is None:
                 raise RuntimeError("""\
-Error populating the input "%s" of node "%s": the results file of the source node \
-(%s) does not contain any outputs.""" % (key, self.name, results_file))
-            output_value = Undefined
-            if isinstance(info[1], tuple):
-                output_name = info[1][0]
-                value = getattr(outputs, output_name)
-                if isdefined(value):
-                    output_value = evaluate_connect_function(
-                        info[1][1], info[1][2], value)
-            else:
-                output_name = info[1]
+Error populating the inpus of node "%s": the results file of the source node \
```

Comment:
```suggestion
Error populating the inputs of node "%s": the results file of the source node \
```

just caught this typo. will fix with your review suggestions.